src/align.cpp

```diff
@@ -1418,18 +1418,19 @@ static CmtAlignType get_comment_align_type(chunk_t *cmt)
 chunk_t *align_trailing_comments(chunk_t *start)
 {
    LOG_FUNC_ENTRY();
-   int          min_col  = 0;
-   int          min_orig = -1;
+   size_t       min_col  = 0;
+   size_t       min_orig = 0;
    chunk_t      *pc      = start;
-   int          nl_count = 0;
+   size_t       nl_count = 0;
    ChunkStack   cs;
    CmtAlignType cmt_type_start, cmt_type_cur;
-   int          col;
-   int          intended_col = cpd.settings[UO_align_right_cmt_at_col].n;
+   size_t       col;
+   size_t       intended_col = cpd.settings[UO_align_right_cmt_at_col].n;
```

Comment:
All number options are ints, some options have special properties when values are below zero. Someone could set the align_right_cmt_at_col to negative values in their config and underflow the size_t. 
If size_t is better suited here a `max(0, cpd.settings[UO_align_right_cmt_at_col].n);` could be used to prevent the underflow.

A thing that I noticed is that min_val and max_val in option_map_value are not used. This values could be used to prevent out of rang values while parsing the config.